### DISCUSSION TITLE
Long commands get split into 2 lines

### DISCUSSION THREAD

**Discussion by Jai-JAP** [akinomyoga/ble.sh#310] [Q&A]
**ble version**: 0.4.0-devel3+2eadcd5
**Bash version**: 5.1.16(1)-release (x86_64-pc-linux-gnu)

```
GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu) [Ubuntu 22.04.2 LTS]
ble.sh, version 0.4.0-devel3+2eadcd5 (noarch) [git 2.34.1, GNU Make 4.3, GNU Awk 5.1.0, API: 3.0 (GNU MPFR 4.1.0, GNU MP 6.2.1)]
bash-completion, version 2.11 (hash:b42f5d6a7ad6d4921ec73838ba54a96d6bd30936, 77071 bytes) (noarch)
locale: LANG=en_IN
terminal: TERM=xterm-256color wcwidth=auto-auto/15.0-2+ri, vte:6800 (65;6800;1)
```

After enabling git-prompt in ble-sh, the command gets split into multiple lines as shown in the below image.

![Screenshot from 2023-03-31 18-13-15](https://user-images.githubusercontent.com/78354625/229126428-6b125e24-2ad2-47b7-afb4-670554f05981.png)

This is my bashrc customisation for interactive shells if it helps.
```
if [[ $- == *i* ]]; then
  if [[ $USER == "root" ]]; then
    PS1='${debian_chroot:+($debian_chroot)}\[\033[01;31m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\] \n \[\033[01;31m\]\$_\[\033[00m\] '
  else
    PS1='${debian_chroot:+($debian_chroot)}\[\033[01;32m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\] \n \[\033[01;32m\]\$_\[\033[00m\] '
  fi
  if [[ "$TERM_PROGRAM" != "vscode" ]]; then
    source ~/.ble-sh/out/ble.sh
  fi
fi
```

This is my blerc
```
ble-bind -f 'C-h' kill-backward-eword
ble-import contrib/prompt-git
bleopt prompt_rps1='\q{contrib/git-info}'
```

**Comment by Jai-JAP** [comment]
My question is that whether its an bug or its an issue with my ble configuration

**Reply by akinomyoga** [reply]
> My question is that whether its an bug or its an issue with my ble configuration

Ah, it's neither. When you have the right prompt, the textarea shrinks. This is the logical layout inside ble.sh:

![image](https://user-images.githubusercontent.com/8982192/229130602-b261546d-ddab-4ed5-b03e-fcc9e31e7d15.png)

The orange part is the prompt, the blue box is the right-prompt area, and the green part is the command line.

**Reply by Jai-JAP** [reply]
Ok. But isn't that a waste of space in situations like this.
Can you not shrink the height of right prompt as required and instead increase command line area? 
Or it that a bit too much l?

**Reply by akinomyoga** [reply]
It might be technically possible, but the idea of the right prompt is to put information on the same line as the command line area, which is impossible to achieve with conventional ways using the PS1. If you would like to keep the right-prompt lines not overlapping with the command line, you could put it in the normal prompt with appropriate spacing.

I once thought about the non-square shape (in other words, arbitrary shape) of the text area, but the implementation could be complicated. Maybe I'll reconsider it.

**Reply by Jai-JAP** [reply]
Can this be implemented? the current implementation looks really bad for loong commands with `prompt_rps1` set.

I am ready to help in any possible way...

**Reply by akinomyoga** [reply]
I just don't have time to spend on it.

**Reply by akinomyoga** [reply]
If you could implement it properly without needing my help, I could review the PR. If one requires help, it's easier for me to do that myself than to explain everything to the user.
